uniffi_meta/src/metadata.rs

```diff
@@ -23,7 +23,7 @@ pub mod codes {
     pub const UNIFFI_TRAIT: u8 = 11;
     pub const TRAIT_INTERFACE: u8 = 12;
     pub const CALLBACK_TRAIT_INTERFACE: u8 = 13;
-    //pub const UNKNOWN: u8 = 255;
+    pub const AS_TRAIT: u8 = 14;
```

Comment:
`AS_TRAIT` doesn't communicate what this is doing to me, I keep thinking of `AsRef<T>` and expecting some sort of casting to be happening.  That's essentially what happens when you use the object as a trait, but here we're just tracking the list.  What about something like `TRAIT_IMPLS`?